crates/bitwarden-crypto/src/store/context.rs

```diff
@@ -168,8 +168,35 @@ impl<Ids: KeyIds> KeyStoreContext<'_, Ids> {
         encryption_key: Ids::Symmetric,
         key_to_encrypt: Ids::Symmetric,
     ) -> Result<EncString> {
-        let key_to_encrypt = self.get_symmetric_key(key_to_encrypt)?;
-        self.encrypt_data_with_symmetric_key(encryption_key, &key_to_encrypt.to_vec())
+        let wrapping_key = self.get_symmetric_key(encryption_key)?;
+        match wrapping_key {
+            // These keys wrap directly by encrypting the key bytes of the inner key, with padding
+            // applied in case it is needed
+            SymmetricCryptoKey::Aes256CbcKey(_) | SymmetricCryptoKey::Aes256CbcHmacKey(_) => {
+                let key_to_encrypt = self.get_symmetric_key(key_to_encrypt)?;
+                self.encrypt_data_with_symmetric_key(encryption_key, &key_to_encrypt.to_encoded())
+            }
+            // These keys wrap using CBOR. The content type needs to indicate what the format of the
+            // inner key is
+            SymmetricCryptoKey::XChaCha20Poly1305Key(k) => {
+                let key_to_encrypt = self.get_symmetric_key(key_to_encrypt)?;
+                match key_to_encrypt {
+                    SymmetricCryptoKey::Aes256CbcKey(_)
+                    | SymmetricCryptoKey::Aes256CbcHmacKey(_) => {
+                        let encoded_key = key_to_encrypt.to_encoded_raw();
```

Comment:
The usage of this raw encoding outside of the module seems pretty risky, to me.

I don't see a way to choose between these methods as an outsider with any degree of confidence.

Reply:
Yeah, the public function of the module can take a required wrapping key type as a parameter and make the decision on how to encode internally.

This might also allow you to clean up some of the documentation on when to use each of these methods, since it'd be clearer in a central, public location.

The context around this key padding is fiddly and pretty weird. I don't expect it to be touched often and because of that feels likely to be mishandled when it is changes. It feels worth making as restrictive and self-evident as possible, to me.